rest/handler.go

```diff
@@ -41,6 +41,17 @@ const (
 	minCompressibleJSONSize = 1000
 )
 
+// Admin API Auth Roles
+const (
+	MobileSyncGatewayRole = "mobile_sync_gateway"
+	BucketFullAccessRole  = "bucket_full_access"
+	FullAdminRole         = "admin"
+	ReadOnlyAdminRole     = "ro_admin"
+)
+
+var BucketScopedEndpointRoles = [...]string{MobileSyncGatewayRole, BucketFullAccessRole, FullAdminRole}
```

Comment:
Is the reason for [...]string vs. []string to do with slice memory allocation?  If so, do we still benefit from that if the only place we use it we're converting it to a regular slice via BucketScopedEndpointRoles[:]?  I'm not sure if the latter results in a new allocation - do you know?

Reply:
It was the closed version I could think of to having a const. Also unsure on the final memory allocation when converting it back to [:]

Reply:
Did a quick benchmark but wasn't able to verify what impact it has on memory allocation. Like above, main reason for the fixed size array was to get as close to a constant as I could.

Reply:
As discussed elsewhere, I think []string is preferable if the only use case is to do [:].